### PR TITLE
lib/uplink: ensure tests are compiled separately

### PR DESCRIPTION
**Pull request by egonelbre**
What: Use `uplink_test` for all tests in `lib/uplink`

Why: The tests depend on testplanet which wants to use `lib/uplink`, and this breaks that cycle.

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:
None

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: pkg/storj/object.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ var (
 // Bucket contains information about a specific bucket
 type Bucket struct {
 	Name                 string
-	Attribution          string
+	Attribution          string // TODO: this should be uuid.UUID
```
Comment (jtolds) [inline pkg/storj/object.go]: these should be user agent formatted strings? not uuids

File: lib/uplink/bucket.go [hunk 2]
```diff
@@ -29,6 +29,9 @@ type Bucket struct {
 
 // TODO: move the object related OpenObject to object.go
 
+// Info returns information about the bucket.
+func (b *Bucket) Info() storj.Bucket { return b.bucket }
```
Comment (jtolds) [inline lib/uplink/bucket.go]: any field we need here should probably be on the bucket type directly? there are already a few

File: pkg/storj/object.go (hunk 1 above)
Comment (zeebo) [reply pkg/storj/object.go]: Not according to everywhere else in the code base. For example, see https://github.com/storj/storj/blob/f9ed0dc1a83c0b7e6debef465ac846af56192885/satellite/attribution/db.go#L18-L24

File: lib/uplink/bucket.go (hunk 2 above)
Comment (zeebo) [reply lib/uplink/bucket.go]: `CreateBucket` on the project type already returns a `storj.Bucket` https://github.com/storj/storj/blob/f9ed0dc1a83c0b7e6debef465ac846af56192885/lib/uplink/project.go#L97

We could promote up just the `Attribution` field, but that seems silly to do just for a test. I'm just going to work around this with some reflect magic, for now.

**Comment by zeebo** [comment]
I hopefully did the minimum required to get this merged.

**Pull request merged**